ddtrace/tracer.py

```diff
@@ -147,20 +154,37 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
             parent = None
 
         if parent:
-            # this is a child span
+            trace_id = parent.trace_id
+            parent_span_id = parent.span_id
+            sampling_priority = parent.get_sampling_priority()
+        else:
+            trace_id, parent_span_id, sampling_priority = context.get_context_attributes()
+
+        if trace_id:
+            # child_of a non-empty context, so either a local child span or from a remote context
+
+            # when not provided, inherit from parent's service
+            if parent:
+                service = service or parent.service
+
             span = Span(
                 self,
                 name,
-                service=(service or parent.service),
+                trace_id=trace_id,
+                parent_id=parent_span_id,
+                service=service,
                 resource=resource,
                 span_type=span_type,
-                trace_id=parent.trace_id,
-                parent_id=parent.span_id,
             )
-            span._parent = parent
-            span.sampled = parent.sampled
+            span.set_sampling_priority(sampling_priority)
```

Comment:
Maybe I'm missing something, but the fact that `sampling_priority` is both in the `Span` and in the `Context` could be a code smell of some primitives that we need but we don't have. This means that the `Context` could have a specific priority, while a nested `Span` another one.

If I have an error and want to give priority 1 (the highest let's say) so that the trace is surely sampled, how can I propagate that to the `Context`?